From e8ca6a0ebe4ae71adfe46c2cde01396f9e5bb0d8 Mon Sep 17 00:00:00 2001
From: Konsta <konsta09@gmail.com>
Date: Mon, 1 Sep 2025 18:52:41 +0300
Subject: [PATCH 3/3] remove AV_CODEC_CAP_SUBFRAMES

* AV_CODEC_CAP_SUBFRAMES has been deprecated [1] in FFmpeg 6.1 and
  removed [2] in FFmpeg 8.0.
* Audio codecs that used this (e.g. WMAPro) were never supported
  on Raspberry Pi ffmpeg_codec2 implementation anyway.

[1]: https://github.com/ffmpeg/ffmpeg/commit/8b20d0dcb5cfa8dc34830acdd719fc65b8b5ef67
[2]: https://github.com/ffmpeg/ffmpeg/commit/c29a1cbd03d5dd6b3161e1acf9cb31111511ac0a
---
 C2FFMPEGAudioDecodeComponent.cpp | 30 ++++--------------------------
 1 file changed, 4 insertions(+), 26 deletions(-)

diff --git a/C2FFMPEGAudioDecodeComponent.cpp b/C2FFMPEGAudioDecodeComponent.cpp
index 7394661f3..9f078efab 100644
--- a/C2FFMPEGAudioDecodeComponent.cpp
+++ b/C2FFMPEGAudioDecodeComponent.cpp
@@ -655,33 +655,11 @@ void C2FFMPEGAudioDecodeComponent::process(
 
             std::shared_ptr<C2Buffer> buffer = createLinearBuffer(std::move(block), 0, len);
 
-            if (mCtx->codec->capabilities & AV_CODEC_CAP_SUBFRAMES) {
-                auto fillWork = [buffer, &work, this](const std::unique_ptr<C2Work>& clone) {
-                    clone->worklets.front()->output.configUpdate = std::move(work->worklets.front()->output.configUpdate);
-                    clone->worklets.front()->output.buffers.clear();
-                    clone->worklets.front()->output.buffers.push_back(buffer);
-                    clone->worklets.front()->output.ordinal = clone->input.ordinal;
-                    if (mFrame->best_effort_timestamp != AV_NOPTS_VALUE) {
-                        work->worklets.front()->output.ordinal.timestamp = mFrame->best_effort_timestamp;
-                    }
-                    clone->worklets.front()->output.flags = C2FrameData::FLAG_INCOMPLETE;
-                    clone->workletsProcessed = 1u;
-                    clone->result = C2_OK;
-                };
-
-#if DEBUG_FRAMES
-                ALOGD("process: send subframe buffer ts=%" PRIu64 " idx=%" PRIu64,
-                      work->input.ordinal.timestamp.peeku(), work->input.ordinal.frameIndex.peeku());
-#endif
-                cloneAndSend(work->input.ordinal.frameIndex.peeku(), work, fillWork);
-            }
-            else {
-                work->worklets.front()->output.buffers.push_back(buffer);
-                if (mFrame->best_effort_timestamp != AV_NOPTS_VALUE) {
-                    work->worklets.front()->output.ordinal.timestamp = mFrame->best_effort_timestamp;
-                }
-                break;
+            work->worklets.front()->output.buffers.push_back(buffer);
+            if (mFrame->best_effort_timestamp != AV_NOPTS_VALUE) {
+                work->worklets.front()->output.ordinal.timestamp = mFrame->best_effort_timestamp;
             }
+            break;
         }
     }
 #if DEBUG_FRAMES
-- 
2.34.1

